chain/beacon/node.go

```diff
@@ -62,27 +64,38 @@ type Handler struct {
 }
 
 // NewHandler returns a fresh handler ready to serve and create randomness
-// beacon
-func NewHandler(c net.ProtocolClient, s chain.Store, conf *Config, l log.Logger, version commonutils.Version) (*Handler, error) {
+// beacon.
+//
+//nolint:lll // This is a complex function
+func NewHandler(ctx context.Context, c net.ProtocolClient, s chain.Store, conf *Config, l log.Logger, version commonutils.Version) (*Handler, error) {
+	ctx, span := metrics.NewSpan(ctx, "NewHandler")
+	defer span.End()
+
 	if conf.Share == nil || conf.Group == nil {
-		return nil, errors.New("beacon: invalid configuration")
+		err := errors.New("beacon: invalid configuration")
+		span.RecordError(err)
```

Comment:
can we just `.RecordError` at the upper scope and get the benefits, or does every function deep in the stack have to record it?